### ISSUE TITLE
Add SleepUntil 

### ISSUE THREAD

**Issue by bbarenblat** [abseil/abseil-cpp#1088]
Feature request: It would be great if Abseil exposed a `SleepUntil` function alongside `SleepFor`. A Linux implementation could be something like this:

```c++
// Copyright 2022 Benjamin Barenblat
// SPDX-License-Identifier: Apache-2.0

// SleepUntil()
//
// Sleeps until the current time is after the specified absolute time.
//
// Notes:
// * Signal interruptions will not reduce the sleep duration.
// * Returns immediately when given a time that has already passed.
void SleepUntil(absl::Time time) {
  struct timespec wake_point = absl::ToTimespec(time);
  while (clock_nanosleep(CLOCK_REALTIME, TIMER_ABSTIME, &wake_point,
                         /*remain=*/nullptr) == EINTR) {
    // Ignore signals and wait for the full interval to elapse.
  }
}
```

This obviously won’t work on Windows, but I believe you can use [`SetWaitableTimer`](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-setwaitabletimer) to achieve something similar. `clock_nanosleep` doesn’t exist on Apple OSes either; I’m not sure what the correct approach is there.

**Comment by bbarenblat** [comment]
Apple OSes do appear to support passing an absolute time to `pthread_cond_timedwait`, but looking at their libpthread, the absolute time gets converted into a relative time before it gets passed to the kernel. My guess is that `pthread_cond_timedwait` won’t wait for the correct amount of time if the system clock gets reset during the wait, but somebody with a Mac should confirm.

**Comment by derekmauro** [comment]
Abseil has `absl::ToChronoTime()`, so you could do
```
void SleepUntil(absl::Time t) {
  std::this_thread::sleep_until(absl::ToChronoTime(t));
}
```

**Comment by bbarenblat** [comment]
This is true. However, both libstdc++ and libc++ implement `sleep_until` int terms of `sleep_for`: They compute the amount of time from now until the requested time point and then sleep for that duration. This has unexpected behavior if the clock has a discontinuity during the sleep (notably, at least on Linux, if the machine enters a suspend state during the sleep). This may be a bug in libstdc++ and libc++, but I’m not sure it’s been discussed. [GCC bug 87679](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87679) is related, but only somewhat, since it’s dealing with a custom clock.

Anyway, even though the standard library has comparable functionality, I still think an Abseil function would be useful. I (and probably others) would appreciate an implementation that is explicitly guaranteed to sleep until the system clock passes a certain point, regardless of discontinuities, sleep, etc. If this does in fact turn out to be a bug in libstdc++ and libc++, Abseil could switch to `sleep_until` later down the line.
